autogpt_platform/backend/backend/server/v2/library/routes/presets.py

```diff
@@ -113,7 +126,12 @@ async def create_preset(
         HTTPException: If an error occurs while creating the preset.
     """
     try:
-        return await db.upsert_preset(user_id, preset)
+        if isinstance(preset, models.LibraryAgentPresetCreatable):
+            return await db.create_preset(user_id, preset)
```

Comment:
This pattern is not neat. We've been catching exceptions through middleware, we can just generally return 404 on any not found exception error that way rather than spamming try catch in all of our endpoint code, similarly with 500.

Reply:
I agree, I'd like to fix this in a different PR though